test/HeadersTest.php

```diff
@@ -424,4 +424,22 @@ public function testHeaderCrLfAttackAddHeadersMultiple()
         $headers->addHeaders(['Fake' => ["foo-bar\r\n\r\nevilContent"]]);
         $headers->forceLoading();
     }
+    
+    public function testCanConvertNonASCIIHeaderToStringAfterAddingHeaderToMessage()
+    {
+        //non ASCII characters
+        $str = 'òàù';
+        $header = new \Zend\Mail\Header\GenericHeader();
+        $message = new \Zend\Mail\Message();
+        $headers = new \Zend\Mail\Headers();
+        
+        $header->setFieldName('X-Test');
+        $header->setFieldValue($str);
+        $headers->addHeader($header);
+        $message->setHeaders($headers);
+        $headers->setEncoding('UTF-8');
+        
+		$this->assertTrue(is_string($header->toString()), 'Can call Header::toString()');
```

Comment:
1. assertTrue(is_Foo) can be replaced with assertInternalType(foo)
2. If test is about an encoding problem is better to test output value is exactly the expected encoded value.